app/controllers/application_controller.rb

```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fwitter"
+  end
+  
+  get '/' do
+    erb :index
+  end
+  
+  get '/signup/?' do
+    if session[:id].nil?
+      erb :'/users/create_user'
+    else
+      redirect to '/tweets'
+    end
+  end
+  
+  get '/login/?' do
+    #binding.pry
+    @user = User.find_by id: session[:id]
+    if @user.nil?
+      #binding.pry
+      erb :'/users/login'
+    else
+      redirect to '/tweets'
+    end
+  end
+  
+  get '/logout' do
+    @user = User.find_by id: session[:id]
+    #binding.pry
+    if !@user.nil?
+      session.clear
+      redirect to '/login'
+    else
+      #binding.pry
+      session.clear
+      redirect to '/'
+    end
+    #binding.pry
+  end
+  
+  get '/tweets' do
+    #binding.pry
+    @user = User.find_by id: session[:id]
+    if !@user.nil?
+      #binding.pry
+      erb :'/tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+  
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    erb :'/tweets/show_user_tweets'
+  end
+  
+  get '/tweets/new' do
+    @user = User.find_by id: session[:id]
+    #binding.pry
+    if !@user.nil?
+      erb :'/tweets/create_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+  
+  get '/tweets/:tweet_id' do
+    if !session[:id].nil?
+      @tweet = Tweet.find_by params[:tweet_id]
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+    #binding.pry
+  end
+  
+  get '/tweets/:tweet_id/edit' do
+    #binding.pry
+    @user = User.find_by id: session[:id]
+    if !@user.nil?
+      @tweet = Tweet.find(params[:tweet_id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?